src/core/public/chrome/global_search/global_search_service.ts

```diff
@@ -96,16 +110,39 @@ export class GlobalSearchService {
     });
   }
 
+  private registerSearchSubmitCommand = (searchSubmitCommand: GlobalSearchSubmitCommand) => {
+    const commands = this.searchSubmitCommands$.getValue();
+    if (commands.find((command) => command.id === searchSubmitCommand.id)) {
+      // eslint-disable-next-line no-console
+      console.warn(`Duplicate SearchSubmitCommands id ${searchSubmitCommand.id} found`);
+      return;
+    }
+    this.searchSubmitCommands$.next([...commands, searchSubmitCommand]);
+  };
+
+  private unregisterSearchSubmitCommand(id: string) {
+    const newCommands = this.searchSubmitCommands$.getValue().filter((item) => {
+      return item.id !== id;
+    });
+    if (newCommands.length === this.searchSubmitCommands$.getValue().length) {
+      return;
+    }
+    this.searchSubmitCommands$.next(newCommands);
+  }
+
   public setup(): GlobalSearchServiceSetupContract {
     return {
       registerSearchCommand: this.registerSearchCommand.bind(this),
+      registerSearchSubmitCommand: this.registerSearchSubmitCommand,
```

Comment:
I don't seem to see where is this function called `registerSearchSubmitCommand`, did I miss anything?